src/Util.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace SilverStripe\Assets;
+
+class Util
+{
+    public static function rewindStream($resource)
+    {
+        if (ftell($resource) !== 0 && static::isSeekableStream($resource)) {
+            rewind($resource);
+        }
+    }
+
+    public static function isSeekableStream($resource)
```

Comment:
```suggestion
    public static function isSeekableStream(resource $resource): void
```

Reply:
DONE 
I cannot add `resource` as a type hint.
See https://wiki.php.net/rfc/scalar_type_hints :
`No type hint for resources is added, as this would prevent moving from resources to objects for existing extensions, which some have already done (e.g. GMP).`

Reply:
Since this is in the Util class, may as well just add the following (adapted from [this](https://stackoverflow.com/a/38430606)):

```
private static function checkIsResource($resource): void
{
    if (!is_resource($resource)) {
        throw new \InvalidArgumentException('$resource argument is not a valid resource');
    }
}
```

And then type check within `rewindStream()` and `isSeekableStream()`

Reply:
DONE